### PR TITLE
test cases complete I hope?

### PR DESCRIPTION
**Pull request by nukenukenukelol**

### CODE REVIEW

**Comment by nukenukenukelol** [comment]
I finish all TODOs in polarCoordinateTraversalTest.m I also add a special test case where ray goes along a ray boundary

File: polarCoordinateTraversalTest.m [hunk 1]
```diff
@@ -43,18 +43,235 @@ function testRayEntersCircleAndGoesThroughOrigin(testCase)
     
     [rVoxels, thetaVoxels] = polarCoordinateTraversal(min_bound, max_bound, ray_origin, ray_direction, ...
         circle_center, circle_max_radius, num_radial_sections, num_angular_sections, t_begin, t_end, verbose);
-    expected_rVoxels     = [1,2,2,2,1];
-    expected_thetaVoxels = [2,2,1,0,0];
```
Comment (cgyurgyik) [inline polarCoordinateTraversalTest.m]: This was incorrect before you changed it?

File: polarCoordinateTraversalTest.m [hunk 2]
```diff
@@ -43,18 +43,235 @@ function testRayEntersCircleAndGoesThroughOrigin(testCase)
...
+    verbose = false;
+    
+    [rVoxels, thetaVoxels] = polarCoordinateTraversal(min_bound, max_bound, ray_origin, ray_direction, ...
+        circle_center, circle_max_radius, num_radial_sections, num_angular_sections, t_begin, t_end, verbose);
+        expected_rVoxels     = [1,2,2,2,1];
+        expected_thetaVoxels = [4,4,3,2,2];
+        
+        verifyEqual(testCase, rVoxels, expected_rVoxels);
+        verifyEqual(testCase, thetaVoxels, expected_thetaVoxels);
+end
```
Comment (cgyurgyik) [inline polarCoordinateTraversalTest.m]: Space in-between functions please

File: polarCoordinateTraversalTest.m [hunk 3]
```diff
@@ -43,18 +43,235 @@ function testRayEntersCircleAndGoesThroughOrigin(testCase)
...
+    verbose = false;
+    
+    [rVoxels, thetaVoxels] = polarCoordinateTraversal(min_bound, max_bound, ray_origin, ray_direction, ...
+        circle_center, circle_max_radius, num_radial_sections, num_angular_sections, t_begin, t_end, verbose);
+        expected_rVoxels     = [2,3,3,3,2,1];
+        expected_thetaVoxels = [2,2,1,1,0,0];
+        
+        verifyEqual(testCase, rVoxels, expected_rVoxels);
+        verifyEqual(testCase, thetaVoxels, expected_thetaVoxels);
+end
```
Comment (cgyurgyik) [inline polarCoordinateTraversalTest.m]: Space in-between functions please

File: polarCoordinateTraversalTest.m [hunk 4]
```diff
@@ -43,18 +43,235 @@ function testRayEntersCircleAndGoesThroughOrigin(testCase)
...
+    
+    [rVoxels, thetaVoxels] = polarCoordinateTraversal(min_bound, max_bound, ray_origin, ray_direction, ...
+        circle_center, circle_max_radius, num_radial_sections, num_angular_sections, t_begin, t_end, verbose);
+        expected_rVoxels     = [1,2,2,1];
+        expected_thetaVoxels = [2,2,3,3];
+        
+        verifyEqual(testCase, rVoxels, expected_rVoxels);
+        verifyEqual(testCase, thetaVoxels, expected_thetaVoxels);
+end
+
```
Comment (cgyurgyik) [inline polarCoordinateTraversalTest.m]: Space in-between functions please

File: polarCoordinateTraversalTest.m [hunk 5]
```diff
@@ -43,18 +43,235 @@ function testRayEntersCircleAndGoesThroughOrigin(testCase)
...
+    verbose = false;
+    
+    [rVoxels, thetaVoxels] = polarCoordinateTraversal(min_bound, max_bound, ray_origin, ray_direction, ...
+        circle_center, circle_max_radius, num_radial_sections, num_angular_sections, t_begin, t_end, verbose);
+        expected_rVoxels     = [2,1];
+        expected_thetaVoxels = [1,1,1];
+        
+        verifyEqual(testCase, rVoxels, expected_rVoxels);
+        verifyEqual(testCase, thetaVoxels, expected_thetaVoxels);
+end
```
Comment (cgyurgyik) [inline polarCoordinateTraversalTest.m]: Space in-between functions please

File: polarCoordinateTraversalTest.m [hunk 6]
```diff
@@ -43,18 +43,235 @@ function testRayEntersCircleAndGoesThroughOrigin(testCase)
...
+    [rVoxels, thetaVoxels] = polarCoordinateTraversal(min_bound, max_bound, ray_origin, ray_direction, ...
+        circle_center, circle_max_radius, num_radial_sections, num_angular_sections, t_begin, t_end, verbose);
+        expected_rVoxels     = [1,2,3,3,2,1];
+        expected_thetaVoxels = [0,0,0,2,2,2];
+        
+        verifyEqual(testCase, rVoxels, expected_rVoxels);
+        verifyEqual(testCase, thetaVoxels, expected_thetaVoxels);
+end
+
+% Special Test. Ray goes along a ray boundary
```
Comment (cgyurgyik) [inline polarCoordinateTraversalTest.m]: Nit: I don't think "Special test" is necessary here. if you want to put something, then "Corner case" seems more appropriate.

File: polarCoordinateTraversalTest.m (hunk 1 above)
Comment (cgyurgyik) [reply polarCoordinateTraversalTest.m]: Screen Shot 2020-02-27 at 6 02 31 PM

Comment (cgyurgyik) [reply polarCoordinateTraversalTest.m]: Here I am seeing:
rVoxels: [1,2,2,2,1]
thetaVoxels: [2,2,1,0,0]

Comment (cgyurgyik) [reply polarCoordinateTraversalTest.m]: Can you explain what you changed for my understanding?

Comment (cgyurgyik) [reply polarCoordinateTraversalTest.m]: Resolved in Slack.

**Comment by cgyurgyik** [comment]
LGTM.
